relay-server/src/actors/project.rs

```diff
@@ -940,9 +940,16 @@ impl Project {
         mut envelope_context: EnvelopeContext,
     ) -> Result<CheckedEnvelope, DiscardReason> {
         let state = self.valid_state();
+        // Treat invalid state as no state at all.
+        // We cannot check anything against invalid state, since it does not contain the required
+        // information, like project id, or quotas, etc.
```

Comment:
If we do this, I think we'll break the v2 endpoint, which relies on an eventual state. Shouldn't we change the places outlined in https://github.com/getsentry/relay/issues/1787#issuecomment-1406473224 instead?